### DISCUSSION TITLE
v1.56.0 - Partner Sharing + Facial Recognition and more! 🎉

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#2468] [Announcements]
# v1.56.0 - Partner Sharing + Facial Recognition and more! 🎉

# Highlights 

Hey there, Immich fans! Summer is here, and I have some exciting news to share!

The team has been working hard over the past few weeks to bring you an amazing release. I couldn't be more thrilled about it, and I hope you'll love it too. We've added two fantastic new features and made several improvements to the app. Let me give you a quick rundown of the highlights:

- Facial Recognition
- Partner Sharing
- Mobile App Lazy Loading
- Logout all devices with a single click

I'm super excited about all these updates, and I hope you are too!

## Facial Recognition

I want to express my gratitude to @LackesLab and @jrasm91 for their invaluable contribution to bringing this feature to life. This feature has been in the works for the past few months, representing one of the final sizable and complex additions to Immich. It fills a crucial gap, bringing Immich closer to feature parity with Google Photos.

We've introduced the ability to recognize people in photos and video thumbnails, cluster assets based on the identified individuals, and showcase them on the Explore page.

![facial-recognition-1](https://github.com/immich-app/immich/assets/27055614/e270afdf-e943-48da-af73-1cbf5a15cf59)

When you click on a person's name, you'll be directed to a view that displays all the assets associated with that person. The returned result is currently limited to 3000 assets due to view limitations. However, our team is actively working on refactoring the timeline to accommodate these scenarios, enabling the loading of all assets related to the selected person.

![facial-recognition-2](https://github.com/immich-app/immich/assets/27055614/6406e6cd-79e4-4a74-bd7b-926d761cf308)

To make things even more convenient, you can assign a name or nickname to each person. And you can use the syntax `m:person-name` in the search bar to find all assets featuring that person effortlessly.

The facial recognition and clustering queues will automatically process all your new uploads. However, for previously uploaded assets, you'll need to head to the `Administration page` and navigate to the `Jobs` section. From there, you can manually run the `Recognize Faces` job for all existing assets.

![image](https://github.com/immich-app/immich/assets/27055614/de11087c-376d-4f09-a489-b4e51afdae54)

**NOTE: This feature has not been implemented on the mobile app**
**NOTE: Anytime you trigger the ALL option to run Recognize Faces, the previous name/nickname that was assigned to the people will be deleted as well**

## Partner Sharing

We extend our appreciation to @brighteyed for bringing one of the most highly requested features to life. With this remarkable addition, Immich has overcome a significant hurdle, eliminating the need for users to share the same account or manually curate photos in shared albums with their partners.

Go to your user settings page to choose who to share with. You can share with as many other users as you like.

<p>
<img src="https://github.com/immich-app/immich/assets/27055614/cccf7f50-bb3c-43c3-922b-f594182cd226" title="partner sharing"> 
<img src="https://github.com/immich-app/immich/assets/27055614/c74e1b89-dd67-4b4d-b642-440162ba46bd" title="partner sharing"> 
</p>

After granting access, users can conveniently view partner libraries under their own Sharing page

![partner-sharing-3](https://github.com/immich-app/immich/assets/27055614/6437de8a-4275-4d3d-807f-f7c6185c53df)

**NOTE: This feature has not been implemented on the mobile app**

## Mobile Lazy Loading

I want to express my gratitude to the continuous contribution efforts of @fyfrey for implementing the lazy loading mechanism in our mobile app. This significant enhancement has resulted in a remarkable improvement in both the browsing experience and the application's overall performance.

With the introduction of lazy loading, users can enjoy a more efficient and optimized browsing experience. By loading content dynamically as it becomes visible, the app minimizes unnecessary data retrieval and rendering, thereby reducing load times and conserving system resources.

Last but not least, we have added many bug fixes and small improvements in the server, web, and mobile clients to make Immich better after each release.

# Contribution

I, Alex, want to thank all the contributors, who have spent countless hours testing, debugging, developing, and brainstorming on GitHub and Discord channels to contribute to Immich. This is a community project, and it cannot happen without you all! THANK YOU!

----

As always, please consider supporting the project.

🎉 Cheer! 🎉

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="Support"> 
</p>

If you find the project helpful and it helps you in some ways, you can support the project [one time](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502) or [monthly](https://github.com/sponsors/alextran1502) from GitHub Sponsors

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

# What changes 

### Mobile
* chore(mobile): Upgrade to Flutter 3.10 by @alextran1502 in https://github.com/immich-app/immich/pull/2429
* feat(mobile): lazy loading of assets by @fyfrey in https://github.com/immich-app/immich/pull/2413

### Server
* feat(server,web): facial recognition by @jrasm91, @alextran1502, @LackesLab in https://github.com/immich-app/immich/pull/2180
* fix(server): return updated asset with correct update time by @alextran1502 in https://github.com/immich-app/immich/pull/2457
* feat (server, web): Share with partners by @brighteyed in https://github.com/immich-app/immich/pull/2388
* feat(web,server): logout all devices by @jrasm91 in https://github.com/immich-app/immich/pull/2415

### Web
* chore(web): update svelte related packages by @michelheusschen in https://github.com/immich-app/immich/pull/2419
* feat(web): album card hover styling by @alextran1502 in https://github.com/immich-app/immich/pull/2424
* feat(web): Replicate albums view for sharing view by @alextran1502 in https://github.com/immich-app/immich/pull/2433
* fix(web): profile image load by @michelheusschen in https://github.com/immich-app/immich/pull/2434
* refactor(web): use ImmichApi to create URLs by @michelheusschen in https://github.com/immich-app/immich/pull/2435
* chore(web): update eslint and prettier packages by @michelheusschen in https://github.com/immich-app/immich/pull/2437
* refactor(web): asset select actions by @michelheusschen in https://github.com/immich-app/immich/pull/2444
* fix(web): use correct favicon sizes by @martabal in https://github.com/immich-app/immich/pull/2446
* fix(web): layout spacing when zooming by @michelheusschen in https://github.com/immich-app/immich/pull/2452
* feat(web): improve page header + scrolling by @jrasm91 in https://github.com/immich-app/immich/pull/2453
* fix(web): unable to change person name by @alextran1502 in https://github.com/immich-app/immich/pull/2458

### Others
* docs: simplify backup dumper example by @jrasm91 in https://github.com/immich-app/immich/pull/2414
* docs: remove roadmap link by @jrasm91 in https://github.com/immich-app/immich/pull/2442
* docs: Add comment about Docker secrets to example.env by @mark-monteiro in https://github.com/immich-app/immich/pull/2454
* chore(ci): versions the Docker cleaning action to latest released by @stumpylog in https://github.com/immich-app/immich/pull/2438
* chore: update api by @alextran1502 in https://github.com/immich-app/immich/pull/2428
* chore: update screenshots for readme and docs by @alextran1502 in https://github.com/immich-app/immich/pull/2425
* chore(ml): use official pytorch channel  by @alextran1502 in https://github.com/immich-app/immich/pull/2416

## New Contributors
* @martabal made their first contribution in https://github.com/immich-app/immich/pull/2446
* @mark-monteiro made their first contribution in https://github.com/immich-app/immich/pull/2454

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.55.1...v1.56.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.56.0'>v1.56.0 - Partner Sharing + Facial Recognition and more! 🎉</a>.</em>

**Comment by LM1LC3N7** [comment]
Wow guys, this is amazing work and app. I am so glad to discover this few weeks ago and would have been found it earlier! 😍

**Comment by CJCutrone** [comment]
Super excited about the facial recognition support! Been looking forward to it since I found the project, excellent work y'all!

I have noticed there are a few "duplicate" people in my library though. It has two groups of face data that are actually the same person. Is there currently a way to combine the two separate face groupings into one, or is that planned for the future?

**Reply by jrasm91** [reply]
Planned, not implemented yet!

**Reply by CJCutrone** [reply]
Nice! Looking forward to it! :D

**Comment by ShinasShaji** [comment]
Amazing work! I have been using Immich for a while now, and it's been amazing so far. Face Recognition is quite accurate and to be honest, PhotoPrism's face recognition just pales in comparison 😅

I've been playing around with it a bit, but is anyone having issues with immich_microservices repeatedly crashing on doing any "Full" job?
 
On realizing that naming duplicate people the same does not merge them at the moment, I tried doing a "Full" Recognize Faces job to remove my face names - which is when I realized this was happening. "Missing" doesn't seem to cause this issue for the time being. 

There are also messages relating to running out of memory in the logs (I have 8 GB allocated to container). I've also had super long statements from immich_postgres dumped into the docker container logs with "FATAL: connection to client lost".

Please let me know if I should start an issue - just checking if anyone else is having the problem.

Edit: Can confirm this still happens in 1.56.1. I shall try allocating more memory to the container and see if that solves the issue later.

Edit 2: It works okay on allocating 16 GB of memory to the container - the memory requirements seem to have increased with this update (I just saw this in the docs as well), since everything used to run okay with just 4 GB of memory earlier.

**Reply by jrasm91** [reply]
The machine learning models use quite a bit of RAM, and at the moment they stay in RAM indefinitely. We're hoping to fix this in the future.

**Reply by ShinasShaji** [reply]
Awesome! Though one thing I noticed is that immich_microservices never really used more than 20-25% of the total memory available to the container - it usually crashes around there. Expanding the total allocated memory allowed immich_microservices to use up to around 3 GB (according to `docker stats`) without crashing. For context, I have around 21k assets in my library.

I'm not very familiar with how containers manage memory, but I felt this behaviour to be odd, I guess.

**Comment by Pheggas** [comment]
Amazing work! Personally, it would be awesome if Immich would ask me if i REALLY want to face recognize all images again as it removes all the labels. There could be scenario that someone who labeled 50+ people will accidentally click the ALL option and the hard time of labeling is lost.

**Reply by jrasm91** [reply]
I think adding a confirm modal there is a good idea.

**Reply by Pheggas** [reply]
I think it wouldn't be hard

**Reply by jrasm91** [reply]
#2496

**Reply by Pheggas** [reply]
Amazing stuff!! Thank y'all!
